setup.py

```diff
@@ -61,6 +61,9 @@
     "dask[dataframe]>=2024.2.1",
     "prometheus_client",
     "psutil",
+    "bigtree>=0.19.2",
+    "pyjwt",
+    "kubernetes<=20.13.0",
```

Comment:
This might require a bit of refactor, but this one definitely shouldn't be a mandatory dependency, especially since we already have an identical k8s extra.

Reply:
Just a small reminder, It was required because the java integration tests were not passing even though kubernetes is already in CI_REQUIRED the java integration tests were still failing.
When you mean refactoring please elaborate

Reply:
The fact that python tests were passing even w/o making them required was because we're doing python tests wrong #4185 :smile: We're installing all dependencies (required + extras) in the env and consequently we have no way of checking whether all required dependencies are actually listed as such or not.

Another way to take care of the failures would be to refactor the code how that kubernetes dependency is no longer top-level, i.e. it will only be imported when `feature_store.yaml` specifies k8s as auth type.

Reply:
This will be addressed in the next/separate PR
cc @tmihalac @lokeshrangineni